addon/initializers/m3-store.js

```diff
@@ -18,12 +18,6 @@ function initializeDebugAdapter(registry) {
 }
 
 export function initialize(application) {
-  // This should be unnecessary
-  // it is done by the meta package
-  // but it should be done by the store package
-  // https://github.com/emberjs/data/issues/7158
-  application.inject('route', 'store', 'service:store');
-  application.inject('controller', 'store', 'service:store');
```

Comment:
This is a breaking change for ember 3.x.  We already have a dep on [ember-compatibility-helpers](https://github.com/ember-cli/ember-compatibility-helpers) so should be able to keep these injections only for ember 3.